app.py

```diff
@@ -353,28 +364,31 @@ def archives():
 
     if category_slug:
         categories = api.get_categories(slugs=[category_slug])
-
-        if categories:
-            category = categories[0]
+        category_ids = list(map(lambda category: category['id'], categories))
+    else:
+        categories = []
+        category_ids = []
 
     posts, total_posts, total_pages = helpers.get_formatted_posts(
         page=page,
         after=after,
         before=before,
         group_ids=[group['id']] if group else [],
-        category_ids=[category['id']] if category else [],
+        category_ids=category_ids if category_ids else [],
     )
 
     return flask.render_template(
         'archives.html',
         posts=posts,
         group=group,
-        category=category,
+        category_slug=category_slug if category_slug else None,
+        categories=categories,
         current_page=page,
         total_posts=total_posts,
         total_pages=total_pages,
         friendly_date=friendly_date,
         now=datetime.now(),
+        category_ids=category_ids
```

Comment:
Can this be alphabetical? I realise some other lines aren't but if you could with the lines you are changing.